### PR TITLE
aws-sso-util: fix build without dependency on cmake or rust

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/aws-sso-util.rb [hunk 1]
```diff
@@ -19,22 +19,22 @@ class AwsSsoUtil < Formula
...
   end
 
   resource "aws-error-utils" do
     url "https://files.pythonhosted.org/packages/4e/7b/622c18e41b17935ac72f4f7b8775e18fe6dd6ecca0d1068fd95f5cbd91f9/aws-error-utils-1.3.0.tar.gz"
     sha256 "188159a8897552408dc3545aed55b49a12532cbde841aad0490e2b93a1275cfc"
+
+    # Use poetry_core backend to avoid unnecessary cmake and rust build dependencies.
+    # Remove when release uses aws-error-utils>=2.6.
+    # Backport of https://github.com/benkehoe/aws-error-utils/commit/f23a4bb8703e8ffabf1d008e04495572cc12d4ee
+    patch :DATA
```
Comment (cho-m) [inline Formula/aws-sso-util.rb:37]: One demerit is patch may break `brew bump` logic.

Other options are:
- Add `cmake` (and keep `rust`) dependency. Demerit is these are probably unnecessary.
- Use `poetry` as dependency and manually run build command. Demerit is that we cannot use `virtualenv_install_with_resources` DSL which makes it harder to maintain.

**Review by p-linnane — APPROVED** [review]
`brew bump` seems to be the easiest to deal with if it breaks.

File: Formula/aws-sso-util.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/aws-sso-util.rb:37]: link to [that PR](https://github.com/benkehoe/aws-error-utils/pull/6)

Comment (chenrui333) [reply Formula/aws-sso-util.rb:37]: maybe we should also file a PR to update `aws-sso-uti` to use `aws-error-utils>=2.6`

**Pull request closed without merging**